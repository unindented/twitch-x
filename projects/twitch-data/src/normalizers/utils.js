export function valueOrNull (value) {
  return value || null
}
