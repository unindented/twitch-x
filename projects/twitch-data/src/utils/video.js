import {stringify} from 'qs'
import {USHER_API_URL} from '../constants'

export function getPlaylistUrl ({stream, sig, token}) {
  const query = {
    sig,
    token,
    allow_audio_only: true,
    allow_source: true,
    player: 'twitchweb',
    type: 'any',
    p: Math.floor(Math.random() * 1e7)
  }
  return `${USHER_API_URL}/channel/hls/${stream}.m3u8?${stringify(query)}`
}

export function extractFeedsFromPlaylist (input) {
  const regexp = /#EXT-X-MEDIA:.*NAME="([^"]+)".*\n#EXT-X-STREAM-INF:.*\n(.*)/g
  const result = []

  let match = regexp.exec(input)
  while (match != null) {
    result.push({
      name: match[1],
      url: match[2]
    })
    match = regexp.exec(input)
  }

  return result
}
