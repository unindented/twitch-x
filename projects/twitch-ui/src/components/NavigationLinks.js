import React from 'react'
import styled from 'styled-components'
import NavigationLink from './NavigationLink'
import homeIcon from 'twitch-assets/build/web/icons/home.svg'
import streamsIcon from 'twitch-assets/build/web/icons/streams.svg'
import gamesIcon from 'twitch-assets/build/web/icons/games.svg'
import communitiesIcon from 'twitch-assets/build/web/icons/communities.svg'
import searchIcon from 'twitch-assets/build/web/icons/search.svg'

const NavigationLinksRoot = styled.ul`
  display: flex;
  flex-direction: column;
  padding: 0;
  margin: 0;
  list-style: none;
`

export default function NavigationLinks () {
  return (
    <NavigationLinksRoot>
      <NavigationLink
        href='/'
        icon={homeIcon}
        iconAlt=''
        label='Home'
      />
      <NavigationLink
        href='/streams'
        icon={streamsIcon}
        iconAlt=''
        label='Streams'
      />
      <NavigationLink
        href='/games'
        icon={gamesIcon}
        iconAlt=''
        label='Games'
      />
      <NavigationLink
        href='/communities'
        icon={communitiesIcon}
        iconAlt=''
        label='Communities'
      />
      <NavigationLink
        href='/search'
        icon={searchIcon}
        iconAlt=''
        label='Search'
      />
    </NavigationLinksRoot>
  )
}
