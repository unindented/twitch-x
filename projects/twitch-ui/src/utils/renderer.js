import React from 'react'
import renderer from 'react-test-renderer'
import {ThemeProvider} from 'styled-components'
import themes from '../themes'

export function renderWithTheme (node, theme = 'default') {
  return renderer.create(
    <ThemeProvider theme={themes[theme]}>
      {node}
    </ThemeProvider>
  )
}
