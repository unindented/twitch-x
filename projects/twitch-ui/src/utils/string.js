export function supplant (str, obj) {
  return str.replace(/\{\s*([^{}]+?)\s*\}/g, (str, key) => obj[key])
}
