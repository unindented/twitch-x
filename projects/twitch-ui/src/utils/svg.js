export function rectangle ({width, height, fill}) {
  return `<svg viewBox="0 0 ${width} ${height}" xmlns="http://www.w3.org/2000/svg"><rect fill="${fill}" width="100%" height="100%" /></svg>`
}
