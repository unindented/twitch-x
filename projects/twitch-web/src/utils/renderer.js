import ShallowRenderer from 'react-test-renderer/shallow'

export function render (node, theme = 'default') {
  const renderer = new ShallowRenderer()
  renderer.render(node)
  return renderer.getRenderOutput()
}
