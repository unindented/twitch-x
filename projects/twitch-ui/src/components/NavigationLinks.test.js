import React from 'react'
import {renderWithTheme} from '../utils/renderer'
import NavigationLinks from './NavigationLinks'

describe('NavigationLinks', () => {
  let tree

  beforeEach(() => {
    tree = renderWithTheme(<NavigationLinks />)
  })

  it('renders', () => {
    expect(tree).toMatchSnapshot()
  })
})
