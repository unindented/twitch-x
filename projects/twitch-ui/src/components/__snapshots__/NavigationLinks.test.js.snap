// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationLinks renders 1`] = `
.c4 {
  display: block;
  width: 50px;
  height: 50px;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  padding: 10px;
}

.c5 {
  display: block;
  width: 100%;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c6 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  margin: 0 10px;
}

.c1 {
  display: block;
}

.c2 {
  display: block;
  color: #fff;
  text-decoration: none;
}

.c2:hover {
  background-color: #392e5c;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 0;
  margin: 0;
  list-style: none;
}

<ul
  className="c0"
>
  <li
    className="c1"
  >
    <a
      className="c2"
      href="/"
    >
      <div
        className="c3"
      >
        <span
          className="c4"
        >
          <img
            alt=""
            className="c5"
            src="file-mock"
          />
        </span>
        <span
          className="c6"
        >
          Home
        </span>
      </div>
    </a>
  </li>
  <li
    className="c1"
  >
    <a
      className="c2"
      href="/streams"
    >
      <div
        className="c3"
      >
        <span
          className="c4"
        >
          <img
            alt=""
            className="c5"
            src="file-mock"
          />
        </span>
        <span
          className="c6"
        >
          Streams
        </span>
      </div>
    </a>
  </li>
  <li
    className="c1"
  >
    <a
      className="c2"
      href="/games"
    >
      <div
        className="c3"
      >
        <span
          className="c4"
        >
          <img
            alt=""
            className="c5"
            src="file-mock"
          />
        </span>
        <span
          className="c6"
        >
          Games
        </span>
      </div>
    </a>
  </li>
  <li
    className="c1"
  >
    <a
      className="c2"
      href="/communities"
    >
      <div
        className="c3"
      >
        <span
          className="c4"
        >
          <img
            alt=""
            className="c5"
            src="file-mock"
          />
        </span>
        <span
          className="c6"
        >
          Communities
        </span>
      </div>
    </a>
  </li>
  <li
    className="c1"
  >
    <a
      className="c2"
      href="/search"
    >
      <div
        className="c3"
      >
        <span
          className="c4"
        >
          <img
            alt=""
            className="c5"
            src="file-mock"
          />
        </span>
        <span
          className="c6"
        >
          Search
        </span>
      </div>
    </a>
  </li>
</ul>
`;
